### PR TITLE
feat(preset-wcag): Add `page-has-valid-lang` rule

### PR DESCRIPTION
**Pull request by wadackel**
## What does this change?

See https://www.w3.org/TR/WCAG20-TECHS/H57.html

## Screenshots

If applicable, add screenshots to help explain your changes.

## What can I check for bug fixes?

Please briefly describe how you can confirm the resolution of the bug.

## References

- https://www.w3.org/TR/WCAG20-TECHS/H57.html

### CODE REVIEW

File: packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import type { ElementHandle } from 'puppeteer-core';
+import { createRule } from '@acot/core';
+import tags from 'language-tags';
```
Comment (wadackel) [inline packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts:3]: I used the `laungage-tags` package to validate the lang attribute.

File: packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+type Options = {};
+
+export default createRule<Options>({
+  type: 'global',
+  immutable: true,
+  meta: {
+    recommended: true,
+  },
```
Comment (wadackel) [inline packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts:12]: This rule is immutable and is included in recommended.

File: packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  test: async (context) => {
+    const html: ElementHandle<HTMLElement> | null = await context.page.$(
+      'html',
+    );
+
+    if (html == null) {
+      return await context.report({
+        message: 'html element MUST be exists.',
+      });
+    }
```
Comment (wadackel) [inline packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts:23]: > 1. Examine the html element of the document.
> https://www.w3.org/TR/WCAG20-TECHS/H57.html#H57-procedure

File: packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      (el) => el.getAttribute('lang') || el.getAttribute('xml:lang'),
+    );
+
+    if (!lang) {
+      return await context.report({
+        node: html,
+        message:
+          'html element MUST have the "lang" or "xml:lang" attribute specified.',
+      });
+    }
```
Comment (wadackel) [inline packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts:35]: > 2. Check that the html element has a lang and/or xml:lang attribute.
> https://www.w3.org/TR/WCAG20-TECHS/H57.html#H57-procedure

File: packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    const valid = tags.check(lang);
+    context.debug('lang="%s"', lang, valid);
+
+    if (!valid) {
+      return await context.report({
+        node: html,
+        message: `"${lang}" lang attribute specified for the html element is invalid. The value of the lang attribute MUST be conforms to BCP 47: Tags for the Identification of Languages or its successor.`,
+      });
+    }
```
Comment (wadackel) [inline packages/acot-preset-wcag/src/rules/page-has-valid-lang.ts:45]: > 3. Check that the value of the lang attribute conforms to BCP 47: Tags for the Identification of Languages or its successor and reflects the primary language used by the Web page.
> https://www.w3.org/TR/WCAG20-TECHS/H57.html#H57-procedure

File: packages/acot-preset-wcag/docs/rules/templates/empty.html [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<!DOCTYPE html>
+{{{ html }}}
```
Comment (wadackel) [inline packages/acot-preset-wcag/docs/rules/templates/empty.html:2]: I added custom template for documentation testing `html` element.

**Comment by masuP9** [comment]
LGTM!! Very careful work!

File: packages/acot-preset-wcag/docs/rules/img-has-name.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # img-has-name
 
-img element or img role MUST has name.
+The `img` element or img role MUST has name.
```
Comment (masuP9) [inline packages/acot-preset-wcag/docs/rules/img-has-name.md:3]: Thanks 😊

**Review by masuP9 — APPROVED** [review]

**Pull request merged**